File: BackEnd/src/main/java/edu/icet/service/custom/CustomerService.java
package edu.icet.service.custom;

import edu.icet.dto.Customer;

import java.util.List;

public interface CustomerService{

    void add(Customer customer);

    List<Customer> getAll();

    void delete(Integer id);

    void updateCustomer(Customer customer);

    Customer searchCustomerById(Integer id);

    List<Customer> searchCustomerByName(String name);

    List<Customer> searchCustomerByAddress(String address);

    List<Customer> searchCustomerBySalary(Double salary);
}

File: FrontEnd/src/app/customer.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Customer } from './models/Customer';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  
  private baseUrl = "http://localhost/8080/customer";

  constructor(private http:HttpClient) { }

  searchCustomerById(id:number): Observable<Customer>{
    return this.http.get<Customer>(`${this.baseUrl}/search-by-id/${id}`);
  }

  searchCustomerByName(name:string): Observable<Customer[]>{
    return this.http.get<Customer[]>(`${this.baseUrl}/search-by-name/${name}`);
  }

  searchCustomerByAddress(address:string): Observable<Customer[]>{
    return this.http.get<Customer[]>(`${this.baseUrl}/search-by-address/${address}`);
  }

  searchCustomerBySalary(salary:number): Observable<Customer[]>{
    return this.http.get<Customer[]>(`${this.baseUrl}/search-by-salary/${salary}`);
  }
}
